compiler/rustc_codegen_ssa/src/codegen_attrs.rs

```diff
@@ -823,6 +823,31 @@ fn autodiff_attrs(tcx: TyCtxt<'_>, id: DefId) -> Option<AutoDiffAttrs> {
         }
     };
 
+    let width: u32;
+    match width_meta {
+        MetaItemInner::MetaItem(MetaItem { path: p1, .. }) => {
```

Comment:
Why is there two different syntaxes for the same thing?

Reply:
I'll double-check, but iirc it's because the width could be 
a) an int lowered from a user calling autodiff by specifying a width (`#[autodiff(df, Forward, 8, args)]`)
b) us adding a dummy 1 to normalize the input if the user doesn't give one (`#[autodiff(df, Forward, args)]`)
In theory I should probably try to match a) even with my dummy width of 1.

Reply:
(Resolved just to make gh happy, lmk if you want to resolve it in this PR).